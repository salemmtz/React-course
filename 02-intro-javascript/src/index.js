const active = true;

const msg = active && 'Active';

console.log(msg);
