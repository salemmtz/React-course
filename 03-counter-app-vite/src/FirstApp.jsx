import PropTypes from 'prop-types';

const newMsg = {
  name: 'Salem',
  msg: 'Hello world',
};

const getResult = () => 4 + 4;

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* Objects have to be convert to be able to display in the DOM */}
      {/* <h1>{JSON.stringify(newMsg)}</h1> */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{subtitle}</p>h1
      <p>{name}</p>
      {/* <p>{getResult()}</p> */}
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};
FirstApp.defaultProps = {
  name: 'Salem Martinez',
  subtitle: 'No subtitle',
  // title: 'No title',
};
