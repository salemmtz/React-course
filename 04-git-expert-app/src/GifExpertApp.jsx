import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch']);

  const onAddCategory = newCategory => {
    if (categories.includes(newCategory)) return;

    // One way to add new categories
    setCategories([newCategory, ...categories]);
    // Another way to add new categories
    // setCategories(cat => [...cat, 'Valorant']);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory
        // setCategories={setCategories}
        onNewCategory={onAddCategory}
      />

      {categories.map(category => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
