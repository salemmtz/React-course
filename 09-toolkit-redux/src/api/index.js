export * from './pokemonAPI';
export * from './todosApi';
