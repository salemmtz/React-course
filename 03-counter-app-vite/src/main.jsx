import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import './styles.css';
import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    {/* <FirstApp title="Hi, I am Salem" /> */}
    <CounterApp value={100} />
  </React.StrictMode>
);
