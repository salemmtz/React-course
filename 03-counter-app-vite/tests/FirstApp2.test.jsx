import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Test in FirstApp', () => {
  const title = 'Hi, I am Salem';
  const subtitle = 'I am a subtitle';

  test('Should match the snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('should show message "Hi, I am Salem"', () => {
    render(<FirstApp title={title} />);
    // screen.debug(); //to see the object
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should show title in h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('should show subtitle sent by props', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(3);
  });
});
