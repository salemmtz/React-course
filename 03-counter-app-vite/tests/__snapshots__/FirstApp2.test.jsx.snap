// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp Should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi, I am Salem
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    No subtitle
  </p>
  <p>
    No subtitle
  </p>
  h1
  <p>
    Salem Martinez
  </p>
</div>
`;
